### DISCUSSION TITLE
[BUG] Unable to create too many multiple connections

### DISCUSSION THREAD

**Discussion by zAstonx** [pnp/powershell#4118] [Q&A]
Good morning,

To perform some deployment task, I need to connect to multiple SharePoint sites.
I use ```Connect-PnPOnline -Url "xxx" -UseWebLogin``` inside a foreach to do it.
Untill version 2.2.0, I could create hundreds of connections to different sites without any issue.
Starting from version 2.3, after 3-4 connections, the login prompt gets stuck in a white screen and I have to close the window and the script stops.
Everytime you update the module, I test it but I always have to rollback to versione 2.2.0.

I use PowerShell 7.4.4 on Windows 11 Pro and all my teammates are experiencing the same issue.

What did you change to the cmdlet after version 2.2.0?
Does anyone have another way to do those multiple connections in sequence?

**Comment by Nime-X** [comment]
I have exactly the same issue with same pwsh version and Windows 11 (on both Pro and Enterprise versions). Issue occurred on previous pwsh versions as well.

**Comment by FlorianGeiger** [comment]
Did you try Connect-PnPOnline -Url "xxx" **-Interactive** ?

**Comment by zAstonx** [comment]
I use the ```-UseWebLogin``` because it doesn't require my action to complete the login mainly.
With ```-Interactive```, if I do manually (not using a script) multiple connections, it won't show me the login screen (that's great.) but when I use it inside a foreach, it asks the select the account in the login screen at every iteration and it works for 6-7 connections but then a red message saying "We couldn't sign you in. Please try again." above my account in the login screen, so I have to wait 10 minutes before trying again. It seems like SP denies the connections after few fast attempts so there is a cooldown.

**Comment by FlorianGeiger** [comment]
The login screen should actually only appear once with the first connect statement. It sounds like there is a disconnect statement in the foreach loop. If this is the case, comment this out and try again.

**Comment by zAstonx** [comment]
This is an example:

```
#Requires -Version 7 -Modules @{ ModuleName = "PnP.PowerShell"; ModuleVersion = "2.2.0" }

try {
    # File path
    $CSVPath = (Read-Host -Prompt 'CSV Path').Trim('"')
    
    $csv = Import-Csv -Path $CSVPath -Delimiter ';'
    # CSV validation
    $validCols = @('SiteUrl')
    $validCounter = 0
    ($csv | Get-Member -MemberType NoteProperty).Name | ForEach-Object {
        if ($_ -in $validCols) { $validCounter++ }
    }
    if ($validCounter -lt $validCols.Count) {
        Write-Host "Mandatory fields: $($validCols -join ', ')" -ForegroundColor Red
        Exit
    }

    $rowCounter = 0
    Write-Host "Begin..." -ForegroundColor Cyan
    foreach ($row in $csv) {
        if ($csv.Count -gt 1) { Write-Progress -Activity "Deploy" -Status "$($rowCounter+1)/$($csv.Count)" -PercentComplete (($rowCounter++ / $csv.Count) * 100) }

        # Connection
        $mainConn = Connect-PnPOnline -Url $row.SiteUrl -UseWebLogin -ValidateConnection -WarningAction SilentlyContinue -ErrorAction Stop -ReturnConnection
        $siteCode = $row.SiteUrl.Split("/")[-1]
        
        Write-Host "Site: $($siteCode)" -ForegroundColor Blue
    }
}
catch { throw }
finally { if ($csv.Count -gt 1) { Write-Progress -Activity "Deploy" } }
```

I don't know if you can test it, but if you try to run this code with a versione newer than 2.2.0, it will get stuck after 3-4 iterations.

> The login screen should actually only appear once with the first connect statement.

To me, using ```-Interactive``` at every single iteration it asks me to select the account to login. Am I doing anything wrong?

**Comment by Nime-X** [comment]
Same behaviours on my side.
Moreover, not all of us are properly authorized to use -Interactive, that's why most of us use the -UseWebLogin.

**Comment by FlorianGeiger** [comment]
I can not use -UseWebLogin. If I am changing the connection statement to
$mainConn = Connect-PnPOnline -Url $url -Interactive -WarningAction SilentlyContinue -ErrorAction Stop -ReturnConnection 
it works fine for me on 36 sites / connections.

I had also to remove the -ValidateConnection parameter. With that parameter I'm getting the error "Connect-PnPOnline: Host not reachable".
<img width="736" alt="Connections" src="https://github.com/user-attachments/assets/99a33bae-d340-47fc-b3ab-6e28ddaf6f95">

**Comment by zAstonx** [comment]
Using PnP.PowerShell 2.5.0
Using -Interactive
Removed -ValidateConnection

At 2nd iteration is asking once again to login.

![Screenshot 2024-07-25 172735](https://github.com/user-attachments/assets/25975231-2382-436d-91cf-03ebe6accd68)

Is this behavior related to Az.Accounts module?

**Comment by zAstonx** [comment]
Can anyone tell me why do I meet this issue and help me fix it?
